qiskit/algorithms/evolvers/evolution_problem.py

```diff
@@ -17,16 +17,29 @@
 from qiskit import QuantumCircuit
 from qiskit.circuit import Parameter
 from qiskit.opflow import OperatorBase, StateFn
+from qiskit.utils.deprecation import deprecate_function
 from ..list_or_dict import ListOrDict
 
 
 class EvolutionProblem:
-    """Evolution problem class.
+    """Pending deprecation: Evolution problem class.
+
+    The EvolutionProblem class has been superseded by the
+    :class:`qiskit.algorithms.time_evolvers.EvolutionProblem` class.
```

Comment:
```suggestion
    :class:`qiskit.algorithms.time_evolvers.TimeEvolutionProblem` class.
```
As ll the replacement classes now have "Time" added to them as part of the name the replacements' names need to be adjusted. I am just commenting in this one place/one module but its needs doing in several places.